cpp/src/arrow/table.h

```diff
@@ -208,6 +208,15 @@ class ARROW_EXPORT Table {
   Result<std::shared_ptr<Table>> CombineChunks(
       MemoryPool* pool = default_memory_pool()) const;
 
+  /// \brief Make a new record batch by combining the chunks this table has.
+  ///
+  /// All the underlying chunks in the ChunkedArray of each column are
+  /// concatenated into a single chunk.
+  ///
+  /// \param[in] pool The pool for buffer allocations
+  Result<std::shared_ptr<RecordBatch>> CombineChunksToBatch(
```

Comment:
There is already a ConcatenateTables which combines multiple tables into a single tables.  Also, there is already a CombineChunks which concatenates the underlying arrays but leaves it as a table. Since the behavior is more similar to the latter I went with that naming scheme.